### DISCUSSION TITLE
Old vs new VAD for 8 kHz wav file (wind noise)

### DISCUSSION THREAD

**Discussion by garymmi** [snakers4/silero-vad#141] [Q&A] [answered]
I have a 8k wav file , the wav file is a kind of wind cut noise
the noise file is at here. [8k_wind_cut_noise](https://drive.google.com/file/d/10qS5uf85sBBPSzO2DNvaI8PSqtRw9Gme/view?usp=sharing)

if use old vad to test, it's ok , no vad result
but use new vad to test, it will output 5 segments. But my expectation is no vad output

code snippet

```
SAMPLING_RATE=8000
model, utils = torch.hub.load(repo_or_dir='snakers4/silero-vad',
                                 model='silero_vad',
                                 force_reload=False,
                                 onnx=USE_ONNX)
(get_speech_timestamps,
 save_audio,
 read_audio,
 VADIterator,
 collect_chunks) = utils
wavfile='Test8k_8_wind_cut.wav'
wav = read_audio(wavfile, sampling_rate=SAMPLING_RATE)
speech_timestamps = get_speech_timestamps(wav, model, sampling_rate=SAMPLING_RATE, visualize_probs=True, return_seconds=True)
pprint(speech_timestamps)
```

output log
Using cache found in /home/gary/.cache/torch/hub/snakers4_silero-vad_master
/home/gary/.cache/torch/hub/snakers4_silero-vad_master/utils_vad.py:184: UserWarning: window_size_samples is too big for 8000 sampling_rate! Better set window_size_samples to 256, 512 or 1536 for 8000 sample rate!
  warnings.warn('window_size_samples is too big for 8000 sampling_rate! Better set window_size_samples to 256, 512 or 1536 for 8000 sample rate!')
PuTTY X11 proxy: unable to connect to forwarded X server: Network error: Connection refused
[{'end': 5.6, 'start': 5.2},
 {'end': 7.5, 'start': 6.1},
 {'end': 10.2, 'start': 8.4},
 {'end': 11.6, 'start': 11.1},
 {'end': 16.9, 'start': 15.6}]

prob. graph 
![image](https://user-images.githubusercontent.com/56832974/146873827-fd219cf9-1133-4dc2-9d5f-84d88fcd1b2d.png)

how to solve this problem?
thanks

**Comment by snakers4** [comment, marked as answer]
Hi,

> warnings.warn('window_size_samples is too big for 8000 sampling_rate! Better set window_size_samples to 256, 512 or 1536 for 8000 sample rate!')

Hm, maybe you forgot to alter the window size for 8 kHz?

**Reply by snakers4** [reply]
I fiddled with your file a bit, fixed a small typo in a warning message and came up with these params:

```
SAMPLING_RATE = 8000
wav = read_audio('Test8k_8_wind_cut.wav', sampling_rate=SAMPLING_RATE)
# get speech timestamps from full audio file
speech_timestamps = get_speech_timestamps(wav,
                                          model,
                                          sampling_rate=SAMPLING_RATE,
                                          window_size_samples=768,
                                          return_seconds=True,
                                          visualize_probs=True,
                                          threshold=0.9,
                                          min_speech_duration_ms=1000)
pprint(speech_timestamps)
# output
# []
```

![image](https://user-images.githubusercontent.com/12515440/146876434-88339d88-54ad-4e18-bec7-7f22ca6963e0.png)

- First of all `min_speech_duration_ms=1000` helps suppress spurious activations
- `threshold=0.9` is a reasonable high value for noisy data (also maybe there is some microphone noise in the training data)
- `window_size_samples=768` is the max recommended chunk size for 8 kHz

Hope this helps.

**Comment by garymmi** [comment]
thanks for your quickly reply
I am curious why noise still has high probability ?
It means clean and noisey environment have different thresholds
The thresold must be manually adjusted by environment

**Reply by snakers4** [reply]
> The threshold must be manually adjusted by environment

Typically threshold should be set once per domain.
But actually finding it is a matter of several experiments.

> I am curious why noise still has high probability ?

Well, the stupid answer is that the new VAD tends to have its confidence either around 0 - 0.2 or around 0.9 - 0.95.
And while for clean data 0.5 is a no-brainer, for noisier data 0.9 is better.

The long answer is that in the data pre-processing process microphone noise was probably associated with speech ... because people use microphones to talk =) And probably they talk also in windy conditions. I have to check it carefully, but it looks like the gusts of wind are associated with "speech".
